syntax = "proto3";


package FileServer;

service FileServerService {
  rpc fileReadWriteRequestHandler (FileReadWriteRequest) returns (FileReadWriteResponse) {}
}

/* The request message containing information for file access
 we added re_ipaddr for server to send responce back on another channel.
 One client can send mutiple request hence using request id
*/

message FileReadWriteRequest {
  enum RequestType {
	READ = 0;
	WRITE = 1;
	DELETE  = 2;
  }
  RequestType type = 1;
  uint32 StartByte = 2;
  uint32 EndByte = 3;
  uint64 RequestId = 4;  
  string FileName = 5;
  string ReqIpaddrPort = 6;
  bytes  data = 7;         //This will be filled while writing file to the server
  uint32 stripwidth = 8;
}

/* The response message containing request status and token if success 
*/
message  FileReadWriteResponse {
  enum Status {
	OK = 0;
	ERROR = 1;
  }
  uint64 RequestId = 1;
  Status  ReqStatus = 2;
  bytes  data = 3;
}
